Tests/MongoSwiftTests/ChangeStreamTests.swift

```diff
@@ -14,153 +108,232 @@ final class ChangeStreamTest: MongoSwiftTestCase {
-            expect(res1?.fullDocument?["x"]).to(bsonEqual(1))
-
-            // test that the resumeToken is updated
-            expect(changeStream.resumeToken).to(equal(res1?._id))
-
-            try coll.updateOne(filter: ["x": 1], update: ["$set": ["x": 2] as Document], session: session)
-            let res2 = changeStream.next()
-
-            // test that the change stream contains a change document for the `update` operation.
-            expect(res2).toNot(beNil())
-            expect(res2?.operationType).to(equal(.update))
-            expect(res2?.fullDocument?["x"]).to(bsonEqual(2))
-
-            // test that the resumeToken is updated
-            expect(changeStream.resumeToken).to(equal(res2?._id))
-
-            // test that the change stream contains a change document for the `find` operation.
-            try coll.findOneAndDelete(["x": 2], session: session)
-            let res3 = changeStream.next()
-            expect(res3).toNot(beNil())
-            expect(res3?.operationType).to(equal(.delete))
-
-            // test that the resumeToken is updated
-            expect(changeStream.resumeToken).to(equal(res3?._id))
+
+        // test that the change stream works on a collection when using withEventType
+        let collChangeStream = try coll.watch(withEventType: MyEventType.self)
+        let doc: Document = ["_id": 1, "x": 1, "y": 2]
+        try coll.insertOne(doc)
+        let collChange = collChangeStream.next()
+        expect(collChangeStream.error).to(beNil())
+        expect(collChange).toNot(beNil())
+        expect(collChange?.operation).to(equal("insert"))
+        expect(collChange?.fullDocument).to(equal(MyFullDocumentType(id: 1, x: 1, y: 2)))
+
+        guard try client.serverVersion() >= ServerVersion(major: 4, minor: 0) else {
+            print("Skipping test case for server version \(try client.serverVersion())")
+            return
+        }
+
+        // test that the change stream works on client and database when using withEventType
+        let clientChangeStream = try client.watch(withEventType: MyEventType.self)
+        let dbChangeStream = try db.watch(withEventType: MyEventType.self)
+        let expectedFullDocument = MyFullDocumentType(id: 2, x: 1, y: 2)
+        try coll.insertOne(["_id": 2, "x": 1, "y": 2])
+        let clientChange = clientChangeStream.next()
+        let dbChange = dbChangeStream.next()
+        expect(clientChangeStream).toNot(beNil())
+        expect(clientChangeStream.error).to(beNil())
+        expect(clientChange?.operation).to(equal("insert"))
+        expect(clientChange?.fullDocument).to(equal(expectedFullDocument))
+        expect(dbChangeStream).toNot(beNil())
+        expect(dbChangeStream.error).to(beNil())
+        expect(dbChange?.operation).to(equal("insert"))
+        expect(dbChange?.fullDocument).to(equal(expectedFullDocument))
+    }
+
+    func testChangeStreamOnWithFullDocumentType() throws {
```

Comment:
is there an extra "on" in this test name?

Reply:
fixed typo